### ISSUE TITLE
memory issues in alignment optimization

### ISSUE THREAD

**Issue by McHaillet** [warpem/miss-alignment#35]
Tilt-series alignment has a patch overlap parameter that influences the number of tiles that can be sampled from the volume space. On EMPIAR-10499, I get the following number of patches with different patch overlap:
* with 0 overlap (72 patches total), the number of patches with the computation graph fits in memory (24 gb GPU) and I can optimize over all patches simultaneously 
* with 0.1 overlap (147 patches) I run into issues

I am already over the reconstructions, its just that I accumulate all the 72 boxes together before calculating the loss and backpropagating. 

My questions:
* optimizing over a smaller number of patches that are grabbed from the total number of positions seems like an obvious solution. I guess those positions should be randomly selected?
* Will this work with the LBFGS optimizer? My initial thought was that it might need a more stable target

**Comment by alisterburt** [comment]
Sorry, I'm confused - why does more patches lead to so much higher memory consumption if you're accumulating gradients?

**Comment by McHaillet** [comment]
Ahhhh, I didn't realise that was a thing. I'm not using gradient accumulation, but I totally should.
Consider this solved until I update that

**Comment by alisterburt** [comment]
Sweet! Probably good to still do them in batches as much as possible (but with gradient accumulation so stability is unaffected) - ideally batch size will be determined from available VRAM and params...

Also not sure if you're using torch.compile for anything yet? Can lead to pretty sweet speedups and reduction in memory costs

**Comment by McHaillet** [comment]
Definitely!

And good point about compile. I'll add that as an issue to not forget about it.

**Comment by McHaillet** [comment]
Fixed now in #28
